chia/cmds/cmd_classes.py

```diff
@@ -0,0 +1,322 @@
+                if origin == collections.abc.Sequence:
+                    if "multiple" not in option_args or not option_args["multiple"]:
+                        raise TypeError("Can only use Sequence with multiple=True")
+                    else:
+                        type_arg = get_args(hints[field_name])[0]
+                        if "default" in option_args and (
+                            not isinstance(option_args["default"], tuple)
+                            or any(not isinstance(item, type_arg) for item in option_args["default"])
+                        ):
+                            raise TypeError(
+                                f"Default {option_args['default']} is not a tuple "
+                                f"or all of its elements are not of type {type_arg}"
+                            )
+                elif "multiple" in option_args:
+                    raise TypeError("Options with multiple=True must be Sequence[T]")
+                elif is_type_SpecificOptional(hints[field_name]):
+                    if "required" not in option_args or option_args["required"]:
+                        raise TypeError("Optional only allowed for options with required=False")
+                    type_arg = get_args(hints[field_name])[0]
+                    if "default" in option_args and (
+                        not isinstance(option_args["default"], type_arg) and option_args["default"] is not None
+                    ):
+                        raise TypeError(f"Default {option_args['default']} is not type {type_arg} or None")
+                elif origin is not None:
+                    raise TypeError(f"Type {origin} invalid as a click type")
+                else:
+                    if hints[field_name] == bytes:
+                        type_arg = HexString()
+                    elif hints[field_name] == bytes32:
+                        type_arg = HexString32()
+                    else:
+                        type_arg = hints[field_name]
+                    if "default" in option_args and not isinstance(option_args["default"], hints[field_name]):
+                        raise TypeError(f"Default {option_args['default']} is not type {type_arg}")
+            else:
+                type_arg = option_args["type"]
+
+            kwarg_names.append(field_name)
+            option_decorators.append(
+                click.option(
+                    *option_args["param_decls"],
+                    type=type_arg,
+                    **{k: v for k, v in option_args.items() if k not in ("param_decls", "is_command_option", "type")},
+                )
+            )
+
+    return _CommandParsingStage(
+        my_dataclass=cls,
+        my_option_decorators=option_decorators,
+        my_subclasses=subclasses,
+        my_kwarg_names=kwarg_names,
+        _needs_context=needs_context,
+    )
+
+
+def _convert_class_to_function(cls: Type[ChiaCommand]) -> SyncCmd:
```

Comment:
I haven't worked through all the refactor details, but this sits a bit funny with me.  This creates a local scope function with a closure on the command parser.  Could there instead be a `_CommandParsingStage.__call__()` defined to make the instance directly callable and to avoid the dynamically created functions and closures?